### PR TITLE
Robot toolkit issue220 panel edge constraints

### LINKED ISSUE #220: Robot_Toolkit: Add support for linear constraints

### PR DESCRIPTION
**Pull request by rwemay**
### NOTE: Depends on 

 ### Issues addressed by this PR

 Closes #220 

 

 ### Test files
https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/Robot_Toolkit/Macro%20Tests/Push?csf=1&e=QlayhJ

 ### Changelog

 ### Additional comments

### CODE REVIEW

**Comment by rwemay** [comment]
Added some scale to the test script - 100 panels, to test performance. Compared between previous panel create methods and new and little loss of performance with no edges assigned (is not entering the 'get edges' COM loop). Assigning supports does slow panel/create considerably, but there is no exposed API method to assign supports to groups/selections of edges, as clarified with Autodesk today. 

performance test

old code no supports

new code no supports

File: Robot_Adapter/Create/Panels.cs [hunk 1]
```diff
@@ -118,7 +80,64 @@ private bool CreateCollection(IEnumerable<Panel> panels)
...
+        /***************************************************/
+
+        private bool HasEdgeSupports(List<Edge> edges)
+        {
+            bool hasSupport = false;
+            foreach (Edge edge in edges)
+            {
+                if (edge.Support != null)
+                {
+                    if (edge.Support.Name != "") hasSupport = true;
```
Comment (IsakNaslundBh) [inline Robot_Adapter/Create/Panels.cs]: Micro optimisation, that shouldn't have any real noticeable impact, but you could return true here rather that storing a bool.

**Comment by IsakNaslundBh** [comment]
Left a minor comment. Everything runs fine though! @rwemay let me know if you want to update or not, if not I am happy with the code as it is. Like the restructuring as well, makes the whole thing a lot cleaner!

File: Robot_Adapter/Create/Panels.cs (hunk 1 above)
Comment (rwemay) [reply Robot_Adapter/Create/Panels.cs]: Thanks! updated, tested on my machine (all ok) and pushed. Worth a quick run through the test on your machine for completeness.

**Pull request merged**